src/variables.jl

```diff
@@ -807,6 +807,46 @@ function _moi_constrain_variable(backend::MOI.ModelLike, index, info)
     end
 end
 
+"""
+    ConstrainedVariables <: AbstractVariable
```

Comment:
`AbstractVariable` is missing a docstring. Readers might get confused about how this relates to a `AbstractVariableRef`.